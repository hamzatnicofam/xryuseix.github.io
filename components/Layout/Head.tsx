/** @format */

import Head from "next/head";

interface Props {
  title?: string;
  description?: string;
}

const Header: React.FC<Props> = ({ title, description }: Props) => {
  const defaultDescription =
    "xryuseixのWebページです.競技プログラミングやセキュリティの分野を勉強しています.";
  const defaultTitle = "xryuseix's HomePage";
  return (
    <Head>
      <title>{title ?? defaultTitle}</title>
      <meta charSet="UTF-8" />
      <meta httpEquiv="content-language" content="ja" />
      <meta name="description" content={description ?? defaultDescription} />
      <link rel="icon" href="/favicon.ico" />
      <meta name="og:title" content={title} />
      <meta property="og:type" content="website" />
      <meta name="twitter:card" content="summary_large_image" />
      {/* TODO: 後ほど作成する */}
      {/* <meta
        property="og:image"
        content="https://avatars.githubusercontent.com/u/7278547?s=200&amp;v=4"
      /> */}
    </Head>
  );
};

export default Header;
