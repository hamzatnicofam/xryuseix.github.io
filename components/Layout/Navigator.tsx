/** @format */

import Link from "next/link";

import "bootstrap/dist/css/bootstrap.min.css";
import { Container, Navbar, Nav, NavDropdown } from "react-bootstrap";
import { MdOpenInNew } from "react-icons/md";

type NavigatorProps = {
  className?: string;
};

const Navigator: React.VFC<NavigatorProps> = (props: NavigatorProps) => (
  <div className="bg-dark">
    <Container>
      <Navbar expand="md" bg="dark" variant="dark">
        <Link href="/">
          <a className="navbar-brand">Home</a>
        </Link>
        <Navbar.Toggle aria-controls="navbarResponsive" />
        <Navbar.Collapse id="navbarResponsive" className="mr-auto">
          <Nav as="ul" className="mr-auto">
            <Nav.Item as="li">
              <Link href="/apps">
                <a className="nav-link">Applications</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <Link href="/achievement">
                <a className="nav-link">Achievement</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <Link href="/account">
                <a className="nav-link">Account</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <Link href="/ctf">
                <a className="nav-link">CTF</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <Link href="/news">
                <a className="nav-link">News</a>
              </Link>
            </Nav.Item>
            <NavDropdown
              title="Slides"
              id="basic-nav-dropdown"
              renderMenuOnMount={true}
            >
              <NavDropdown.Item href="/slides">PDF Viewer</NavDropdown.Item>
              <NavDropdown.Item href="/lt">Single-Page App</NavDropdown.Item>
            </NavDropdown>
            <Nav.Item as="li">
              <Link href="/research">
                <a className="nav-link">Research</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <Link href="/api">
                <a className="nav-link disabled">API</a>
              </Link>
            </Nav.Item>
            <Nav.Item as="li">
              <a
                className="nav-link"
                href="https://github.com/xryuseix/xryuseix.github.io"
                target="_blank"
                rel="noopener noreferrer"
              >
                Repository
                <MdOpenInNew />
              </a>
            </Nav.Item>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </Container>
  </div>
);

export default Navigator;
