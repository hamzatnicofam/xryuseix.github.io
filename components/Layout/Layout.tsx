/** @format */

import { FC } from "react";
import Navigator from "./Navigator";
// import Footer from './Footer'

import styles from "../../styles/Layout.module.css";

export const Layout: FC<{
  headerChild?: JSX.Element;
  title?: string;
  description?: string;
}> = ({ children, title, description }) => (
  <div>
    <div className={styles.header}>
      <Navigator />
    </div>
    <br />
    <main className={styles.headerPadding}>{children}</main>
    <div style={{ textAlign: "center" }}>
      {/* <Footer /> */}
      Footer
    </div>
  </div>
);
