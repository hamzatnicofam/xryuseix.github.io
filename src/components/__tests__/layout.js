/**
 * @jest-environment jsdom
 */

import React from 'react'
import renderer from 'react-test-renderer'
import Layout from '../layout'
describe('Layout', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Layout location="/" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
