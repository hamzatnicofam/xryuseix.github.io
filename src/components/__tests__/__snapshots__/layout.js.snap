// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    className="layout_header"
  >
    <header
      className="bg-dark"
    >
      <div
        className="container"
      >
        <nav
          className="navbar navbar-expand-md navbar-dark"
        >
          <a
            aria-current="page"
            className="navbar-brand"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <button
            aria-controls="navbarResponsive"
            aria-label="Toggle navigation"
            className="navbar-toggler collapsed"
            onClick={[Function]}
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            aria-expanded={null}
            className="mr-auto navbar-collapse collapse"
            id="navbarResponsive"
          >
            <ul
              className="mr-auto navbar-nav"
              onKeyDown={[Function]}
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/news"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  News
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/apps"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Applications
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/achievement"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Achievement
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/account"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Account
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/ctf"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  CTF
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="https://github.com/xryuseix/xryuseix.github.io"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Repository
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 1 24 24"
                    width="1em"
                  >
                    <path
                      d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                    />
                    <polyline
                      points="15 3 21 3 21 9"
                    />
                    <line
                      x1="10"
                      x2="21"
                      y1="14"
                      y2="3"
                    />
                  </svg>
                </a>
              </li>
              <li
                className="nav-item"
              >
                <form
                  action="/search"
                  id="form1"
                  method="get"
                >
                  <input
                    id="s"
                    name="s"
                    placeholder="xryuseix CTF!"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="text"
                  />
                  <input
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="submit"
                    value="search"
                  />
                </form>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </header>
    <div
      className="layout_header-button"
    >
      <div
        className="layout_image-mask"
      />
    </div>
  </div>
  <br />
  <div
    className="global-wrapper"
    data-is-root-path={false}
  >
    <main />
  </div>
  <div
    style={
      Object {
        "text-align": "center",
      }
    }
  >
    <footer>
      <img
        alt="アクセスカウンター"
        border="0"
        src="https://www.f-counter.net/j/45/1613887388/"
        style={
          Object {
            "display": "none",
          }
        }
      />
      © 
      2021
      , Ryusei Ishikawa All Right Reserved.
    </footer>
  </div>
</div>
`;
